emgapianns/management/commands/clean_study.py

```diff
@@ -18,105 +18,151 @@
+                pipeline__release_version=pipeline_version,
+            )
 
             if downloads:
-                logger.info("Found {} downloads for job: {}".format(len(downloads), job))
+                logger.info(
+                    "Found {} downloads for analysis job: {}".format(
+                        len(downloads), job
+                    )
+                )
                 logger.info("Removing the following files:")
                 [logger.info(d.realname) for d in downloads]
-                [d.delete() for d in downloads]
+                downloads.delete()
                 logger.info("Removed.")
             else:
                 logger.info("No files to remove.")
 
-            taxonomy_annotations = AnalysisJobTaxonomy.objects.\
-                filter(job_id=job.job_id, pipeline_version=pipeline_version)
+            if not options.get("mongo_tax"):
+                continue
+
+            logger.info("Getting mongo taxonomic annotations.")
 
-            logger.info("Getting mongo annotations.")
+            taxonomy_annotations = AnalysisJobTaxonomy.objects.filter(
+                job_id=job.job_id, pipeline_version=pipeline_version
+            )
 
             if taxonomy_annotations:
                 for tax_ann in taxonomy_annotations:
-                    if download_group == "ITS":
-                        its_onedb = tax_ann.taxonomy_itsonedb
-                        if its_onedb and len(its_onedb):
-                            logger.info("Removing {} ITS ONE annotations from {}".format(len(its_onedb), job))
-                            tax_ann.taxonomy_itsonedb = []
-
-                        its_unite = tax_ann.taxonomy_itsunite
-                        if len(its_unite):
-                            logger.info("Removing {} ITS Unite annotations from {}".format(len(its_unite), job))
-                            tax_ann.taxonomy_itsunite = []
-
-                    if download_group == "rRNA":
-                        ssu = tax_ann.taxonomy_ssu
-                        if ssu and len(ssu):
-                            logger.info("Removing {} SSU annotations from {}".format(len(ssu), job))
-                            tax_ann.taxonomy_ssu = []
-
-                        lsu = tax_ann.taxonomy_lsu
-                        if lsu and len(lsu):
-                            logger.info("Removing {} LSU annotations from {}".format(len(lsu), job))
-                            tax_ann.taxonomy_lsu = []
+
+                    its_onedb = tax_ann.taxonomy_itsonedb
+                    if its_onedb and len(its_onedb):
+                        logger.info(
+                            "Removing {} ITS ONE annotations from {}".format(
+                                len(its_onedb), job
+                            )
+                        )
+                        tax_ann.taxonomy_itsonedb = []
+
+                    its_unite = tax_ann.taxonomy_itsunite
+                    if len(its_unite):
```

Comment:
if its_unite and len(its_unite)?

Reply:
Good catch, I think that could be None. I'll check